pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go

```diff
@@ -135,33 +141,49 @@ func (og *operationGenerator) GenerateRegisterPluginFunc(
 func (og *operationGenerator) GenerateUnregisterPluginFunc(
 	socketPath string,
 	pluginHandlers map[string]cache.PluginHandler,
+	pathToHandlers map[string]cache.HandlerWithName,
 	actualStateOfWorldUpdater ActualStateOfWorldUpdater) func() error {
 
 	unregisterPluginFunc := func() error {
-		client, conn, err := dial(socketPath, dialTimeoutDuration)
-		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- dial failed at socket %s, err: %v", socketPath, err)
+		var pair cache.HandlerWithName
+		var handler cache.PluginHandler
+		ok := false
+		if len(pathToHandlers) != 0 {
+			pair, ok = pathToHandlers[socketPath]
 		}
-		defer conn.Close()
+		if !ok {
```

Comment:
its not clear why we do all this logic if "not ok", what does "not ok" mean in this context, can this variable be more descriptive